src/rust/ide/view/src/documentation.rs

```diff
@@ -133,22 +147,49 @@ impl ViewModel {
             //              https://github.com/enso-org/enso/issues/1063
             let processed = string.replace("\\n", "\n").replace("\"", "");
             let output = match input_type {
-                InputFormat::AST  => parser.generate_html_docs(processed),
+                InputFormat::AST       => parser.generate_html_docs(processed),
                 InputFormat::Docstring => parser.generate_html_doc_pure(processed),
             };
             let output = output?;
-            if output.is_empty() {
-                Ok(PLACEHOLDER_STR.into())
-            } else {
-                Ok(output)
-            }
+            if output.is_empty() { Ok(PLACEHOLDER_STR.into()) }
+            else                 { Ok(output)                 }
         }
     }
 
     /// Create a container for generated content and embed it with stylesheet.
     fn push_to_dom(&self, content:String) {
         let data_str = format!(r#"<div class="docVis">{}{}</div>"#,documentation_style(),content);
-        self.dom.dom().set_inner_html(&data_str)
+        self.dom.dom().set_inner_html(&data_str);
+    }
+
+    /// Append listeners to copy buttons in doc to enable copying examples.
+    /// It is possible to do it with implemented method, because get_elements_by_class_name
+    /// returns top-to-bottom sorted list of elements, as found in:
+    /// https://stackoverflow.com/questions/35525811/order-of-elements-in-document-getelementsbyclassname-array
+    #[allow(unused_must_use)]
+    fn set_listeners_to_copy_buttons(&self) {
+        let code_blocks  = self.dom.dom().get_elements_by_class_name("CodeBlock");
+        let copy_buttons = self.dom.dom().get_elements_by_class_name("copyCodeBtn");
+        let closures     = (0..copy_buttons.length()).map(|i| -> Option<CodeCopyClosure> {
+            let copy_button = copy_buttons.get_with_index(i)?.dyn_into::<HtmlElement>().ok()?;
+            let code_block  = code_blocks.get_with_index(i)?.dyn_into::<HtmlElement>().ok()?;
+            let closure     = move |_event: MouseEvent| {
+                let inner_code = code_block.inner_text();
+                clipboard::write_text(inner_code);
+            };
+            let closure  = Closure::wrap(Box::new(closure).into_fn_mut());
+            let callback = closure.as_ref().unchecked_ref();
+            copy_button.add_event_listener_with_callback("click", callback).map_err(|err| {
+                error!(&self.logger, "Unable to add event listener to copy button: {err:?}")
+            });
+            Some(closure)
+        }).filter_map(|x| {
+            if x.is_none() {
+                error!(&self.logger, "Tried to add event listener to a non-existent copy button.")
+            }
+            x
+        }).collect::<Vec<CodeCopyClosure>>();
```

Comment:
I don't like this solution a little bit because in case of many buttons it will just throw a bunch of the same errors. Instead, just print a single error telling that some buttons were not attached properly. You can also mention rthe indexes which failed. To do it just return `Result<usize,CodeCopyClosure>` from the lambda instead of `Option<CodeCopyClosure>`. Then filter it by all errors, collect them to vector and in case the vec is non-empty - print the error and mention all indexes.

Moreover, we try not doing such big centipede functions - just split them into more variables instead :D